### ISSUE TITLE
Better error messages when a promise never resolves which causes cy.then to timeout

### ISSUE THREAD

**Issue by jennifer-mann** [cypress-io/cypress#108]
Currently if we have a promise inside of a `cy.then` which does not resolve we don't get a good error message.

<img width="594" alt="screen shot 2016-02-18 at 6 31 11 pm" src="https://cloud.githubusercontent.com/assets/1271364/13161767/e01dc048-d66d-11e5-942e-171a9c0c935d.png">

```coffeescript
it.only "goes back to projects view on cancel [00q]", ->
  cy
    .contains(".btn", "Cancel").click().then ->
      @ipc.handle("get:project:paths", null, [])
```

**Comment by jennifer-mann** [comment]
This is fixed in the upcoming `0.14.0` release.

**Comment by brian-mann** [comment]
Fixed in [`0.14.0`](https://github.com/cypress-io/cypress/wiki/Changelog#0140-03082016).
